core/lib/network-recorder.js

```diff
@@ -306,6 +286,33 @@ class NetworkRecorder extends RequestEventEmitter {
     // get out the list of records & filter out invalid records
     const records = networkRecorder.getRawRecords().filter(record => record.isValid);
 
+    // Set record.source to the Crdp target type that requested the resource,
+    // based on the what session the network events came from.
+    {
+      /** @type {Map<string, 'page'|'iframe'|'worker'>} */
+      const sessionIdToTargetType = new Map();
+      for (const message of devtoolsLog) {
+        if (message.method === 'Target.attachedToTarget') {
```

Comment:
Just to check my understanding...

We won't see a `Target.attachedToTarget` event for the default session, but there is a fall-back to the `page` type on L312 anyway. All other sessions should have a `Target.attachedToTarget` event in the DT log.

Reply:
correct

Reply:
Neat, could you add a comment explaining it?

Reply:
timespans will miss the `Target.attachedToTarget` events too, though, won't they?

Reply:
I think you're right, there could be more than 1 missing `Target.attachedToTarget` event in timespan mode. Are you saying we add the `Target.attachedToTarget` events in ourselves?

Reply:
Is the latest commit what you had in mind?

I still need to write a timespan test that would exercise this